lib/utils/report.mjs

```diff
@@ -47,10 +49,11 @@ export default function report(problem) {
 		);
 	}
 
-	const range = node?.rangeBy({ index, endIndex }) ?? {};
+	const range = nodeRangeBy(node, { index, endIndex });
```

Comment:
[question] Does it make sense to make the call of `nodeRangeBy()` lazy until it's needed actually, in order to avoid extra calculation?

Reply:
@ybiquitous see #8083

Reply:
This PR conflicts with #8083. It seems better to me to close #8083 and take the #8083 change into this PR.

Reply:
I agree that it should be called lazily.

But I find the current shape of the code difficult when trying to do so in a robust manner. I find it hard to be sure that each function is using the correct values and that nothing is accidentally using the original inputs as found in the report.

My preferred solution would be to have a dedicated function to normalize all inputs relevant for the source location. I've opened a separate pull request for that idea: https://github.com/stylelint/stylelint/pull/8098

---

I've made it lazily in this pull request as:

```js
	let range;

	if (!line || !start || !end) {
		range = nodeRangeBy(node, { index, endIndex });
	}
```

Reply:
I now understand why you didn't do it in this PR originally.
i.e. with `|| !start || !end` it cannot be considered a refactor anymore